### PR TITLE
Feature/callback for relay

### PR DESCRIPTION
**Pull request by JHolba**
**Purpose**
add a callback for the relay function. This way the frontend can update to pending when we start relaying a message

**Changes**
added a callback returning TransactionResponse

### CODE REVIEW

File: src/proof.ts
Comment (HDauven) [inline src/proof.ts]: We might have to push the callback to before this line. The _wait_ call with no parameter waits for the first confirmation. This confirmation can theoretically take a very long time before the transaction gets included in a block.

In this approach, the frontend/API consumer can start tracking the transaction on its own without having to wait for the library to process the first transaction confirmation.

Comment (HDauven) [reply src/proof.ts]: This way a consumer can start tracking a transactions life cycle independently, even if a case would arise where a user closes their browser or app prematurely to the library wrapping up the calls. It would, for now, be a reasonable solution. 

In the future we can consider splitting the message collecting and relaying. We can than leave it up to the consumer to  handle the relaying per message in a way that fits best with whatever architecture is being used.

**Review by HDauven — CHANGES_REQUESTED** [review]

File: src/proof.ts
Comment (JHolba) [reply src/proof.ts]: thanks. that was actually a bug, yea

**Review by HDauven — APPROVED** [review]
LGTM!

**Pull request merged**
